sdk/storage/azblob/blob/client.go

```diff
@@ -370,14 +337,47 @@ func (b *Client) DownloadToWriterAt(ctx context.Context, writer io.WriterAt, o *
 	return nil
 }
 
-// DownloadToBuffer downloads an Azure blob to a buffer with parallel.
-func (b *Client) DownloadToBuffer(ctx context.Context, _bytes []byte, o *DownloadToBufferOptions) error {
-	return b.DownloadToWriterAt(ctx, shared.NewBytesWriter(_bytes), o)
+// DownloadStream reads a range of bytes from a blob. The response also includes the blob's properties and metadata.
+// For more information, see https://docs.microsoft.com/rest/api/storageservices/get-blob.
+func (b *Client) DownloadStream(ctx context.Context, o *DownloadStreamOptions) (DownloadStreamResponse, error) {
+	downloadOptions, leaseAccessConditions, cpkInfo, modifiedAccessConditions := o.format()
+
+	dr, err := b.generated().Download(ctx, downloadOptions, leaseAccessConditions, cpkInfo, modifiedAccessConditions)
+	if err != nil {
+		return DownloadStreamResponse{}, err
+	}
+
+	offset := int64(0)
+	count := int64(CountToEnd)
+
+	if o != nil && o.Offset != nil {
+		offset = *o.Offset
+	}
+
+	if o != nil && o.Count != nil {
+		count = *o.Count
+	}
+
+	eTag := ""
+	if dr.ETag != nil {
+		eTag = *dr.ETag
+	}
+	return DownloadStreamResponse{
+		b:                          b,
+		BlobClientDownloadResponse: dr,
+		getInfo:                    HTTPGetterInfo{Offset: offset, Count: count, ETag: eTag},
+		ObjectReplicationRules:     deserializeORSPolicies(dr.ObjectReplicationRules),
+	}, err
+}
+
+// DownloadBuffer downloads an Azure blob to a buffer with parallel.
+func (b *Client) DownloadBuffer(ctx context.Context, buffer []byte, o *DownloadBufferOptions) error {
+	return b.downloadWriterAt(ctx, shared.NewBytesWriter(buffer), o)
 }
 
-// DownloadToFile downloads an Azure blob to a local file.
+// DownloadFile downloads an Azure blob to a local file.
 // The file would be truncated if the size doesn't match.
-func (b *Client) DownloadToFile(ctx context.Context, file *os.File, o *DownloadToFileOptions) error {
+func (b *Client) DownloadFile(ctx context.Context, file *os.File, o *DownloadFileOptions) (int64, error) {
```

Comment:
Do we need to return the number of bytes here too?

Reply:
As per the API review discussion, we need to return the size of the file downloaded. And truncate the local file size if required.